### PR TITLE
Rename binaries; Mimic the original repo layout in the distribution

### PR DESCRIPTION
**Pull request by zah**

### CODE REVIEW

**Comment by arnetheduck** [comment]
meh.

**Comment by arnetheduck** [comment]
ie, those are some very long names, and not consistently applied.. if anything, we could switch to build a single `nimbus-eth2` binary that includes both vc and bn, but like this..?

File: docker/dist/entry_point.sh [hunk 1]
```diff
@@ -24,17 +24,20 @@ if [[ -d "${DIST_PATH}" ]]; then
...
-  cp -a ./build/${BINARY} "${DIST_PATH}/"
-  cd "${DIST_PATH}"
+  cp -a ./build/${BINARY} "${DIST_PATH}/build/"
+  cd "${DIST_PATH}/build"
   md5sum ${BINARY} > ${BINARY}.md5sum
   sha512sum ${BINARY} > ${BINARY}.sha512sum
   cd - >/dev/null
 done
 sed -e "s/GIT_COMMIT/${GIT_COMMIT}/" docker/dist/README.md > "${DIST_PATH}/README.md"
-cp -a scripts/makedir.sh docker/dist/run_*_node.sh "${DIST_PATH}/"
```
Comment (stefantalpalaru) [inline docker/dist/entry_point.sh:37]: You should also delete these scripts, since they're no longer used.

**Review by stefantalpalaru — APPROVED** [review]

**Comment by zah** [comment]
I've previously advocated that we should have a single `nimbus` binary that knows how to start specialized processes responsible for syncing with Eth1, the Eth2 beacon chain and specific shards and EEs in the future. The downside of having multiple specialized binaries is the increased size of the distribution (if each one is going to weight dozens of megabytes), while the downsides of a single monolithic binary are the increased build and link times and the larger attack surface.

My initial idea was indeed to use the `nimbus-eth2` name here, but this seemed a little bit shortsighted because this process is likely to be responsible for stateless validation in phase1 and we are likely to introduce another process for participating in the shard network as a stateful or archive node. How is this second process less worthy of carrying the `eth2` name?

**Pull request merged**